whc-shared/src/test/java/com/ibm/whc/deid/shared/pojo/config/masking/BinningMaskingProviderConfigTest.java

```diff
@@ -15,115 +15,128 @@
-    assertNotEquals(config.hashCode(), other.hashCode());
-    other.setFormat("%s");
-    assertTrue(config.equals(other));
-    assertEquals(config.hashCode(), other.hashCode());
-    config.setFormat(null);
-    assertFalse(config.equals(other));
-    assertNotEquals(config.hashCode(), other.hashCode());
-    other.setFormat(null);
-    assertTrue(config.equals(other));
-    assertEquals(config.hashCode(), other.hashCode());
-
-    config.setStartValue(2);
-    assertFalse(config.equals(other));
-    assertNotEquals(config.hashCode(), other.hashCode());
-    other.setStartValue(2);
-    assertTrue(config.equals(other));
-    assertEquals(config.hashCode(), other.hashCode());
-
-    config.setUseStartValue(!config.isUseStartValue());
-    assertFalse(config.equals(other));
-    assertNotEquals(config.hashCode(), other.hashCode());
-    other.setUseStartValue(!other.isUseStartValue());
-    assertTrue(config.equals(other));
-    assertEquals(config.hashCode(), other.hashCode());
-  }
+	@Test
+	public void testValidate() throws Exception {
+		BinningMaskingProviderConfig config = new BinningMaskingProviderConfig();
+		config.validate(null);
+
+		config.setUnspecifiedValueHandling(-1);
+		try {
+			config.validate(null);
+			fail("expected exception");
+		} catch (InvalidMaskingConfigurationException e) {
+			assertEquals("`unspecifiedValueHandling` must be [0..3]", e.getMessage());
+		}
+		config.setUnspecifiedValueHandling(2);
+		config.validate(null);
+
+		config.setBinSize(0);
+		try {
+			config.validate(null);
+			fail("expected exception");
+		} catch (InvalidMaskingConfigurationException e) {
+			assertEquals("`binSize` must be greater than 0", e.getMessage());
+		}
+
+		config.setBinSize(1);
+		config.validate(null);
+
+		config.setBinSize(1000000);
+		config.validate(null);
+
+		config.setBinSize(-1);
+		try {
+			config.validate(null);
+			fail("expected exception");
+		} catch (InvalidMaskingConfigurationException e) {
+			assertEquals("`binSize` must be greater than 0", e.getMessage());
+		}
+		config.setBinSize(5);
+		
+		config.validate(null);
+		config.setUseSingleBucketOverThreshold(true);
+		config.setUseSingleBucketUnderThreshold(true);
+		config.setSingleBucketOverThresholdValue(50.0);
+		config.setSingleBucketUnderThresholdValue(60.0);
+		try {
+			config.validate(null);
+			fail("expected exception");
+		} catch (InvalidMaskingConfigurationException e) {
+			assertTrue(e.getMessage().contains("`singleBucketOverThresholdValue` must be greater than or equal to 'singleBucketUnderThresholdValue'"));
+		}
+	}
+
+	@Test
+	public void testSetFormat() {
+		BinningMaskingProviderConfig config = new BinningMaskingProviderConfig();
+		config.setFormat("abc");
+		assertEquals("abc", config.getFormat());
+		config.setFormat(null);
+		assertEquals(BinningMaskingProviderConfig.DEFAULT_FORMAT, config.getFormat());
+	}
+
+	@SuppressWarnings("unlikely-arg-type")
+	@Test
+	public void testEqualsHashCode() {
```

Comment:
You can delete this method.  We don't need to test the generated hashCode() we have now.  This doesn't check every field anymore, so best to delete it.